clientconn.go

```diff
@@ -779,11 +779,14 @@ func (ac *addrConn) resetTransport(closeTransport bool) error {
 			}
 			ac.mu.Unlock()
 			closeTransport = false
+			timer := time.NewTimer(sleepTime - time.Since(connectTime))
 			select {
-			case <-time.After(sleepTime - time.Since(connectTime)):
+			case <-timer.C:
 			case <-ac.ctx.Done():
+				timer.Stop()
```

Comment:
I find this a little confusing. `timer.C` isn't used after `time.Stop` at all, so there is not point in
```
if !timer.Stop() {
    <-timer.C
}
```
idiom because there is no chance in the occasional spawn of the channel.
I will do this change though if you still think that it's better, though.
Thanks!